smart-contracts/contracts/PublicLock.sol

```diff
@@ -159,7 +168,8 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
     uint _expirationDuration,
     uint _keyPrice,
     uint _maxNumberOfKeys,
-    uint _version
+    uint _version,
+    uint _cancelRefundPenaltyDenominator
```

Comment:
Will this kill compatibility? 
I think it would be ok to only have a small number of "customizable" constructor arguments and then have setters/getters for the things which are less likely to be changed on the constructor, such as `cancelRefundPenaltyPercentage`.

Reply:
Re compatibility: it requires JS updates but otherwise no, this won't break our upgrade process.  (this was the PR I was working on when I thought to bring up the JS wrapper, it makes changes like this and adding a default value easier).

Adding a setter for something like this may be concerning to the Key owners, unless we add some business logic on what the Lock owner may do here.  This is a tough one for me - it's always a trade off - how much control/flexibility do you offer the Lock owner vs how much guarantees/expectations can we keep for users?  A Lock owner changing the penalty after any Key has been purchased, is kind of changing the terms of that previous purchase without any warning.

Reply:
One option for 'business logic' that would make a setter okay is to say the owner can only LOWER the penalty and once it has been lowered, it may never go up again.  We start with a default of say 20% and then they can choose to lower it to 10%, 5%, or 0.  Thoughts?

Reply:
Oh another option may be to allow the Lock owner to make changes up until the first Key is purchased.  So basically `advanced` type options would all require follow up transactions to the initial lock creation.

Reply:
I think this idea of only allow lowering is interesting...

Generally though, as much as I understand the worry about making it "unfair" for the key owner, we should remember that in practice the lock owner may just decide to stop honoring the lock altogether inside their application :/ Making all keys invalid by definition.

So, in the end, I am not fully sure that limiting the ability to change the cancellation terms is a very useful thing which may backfire in something much worse.

Reply:
> in practice the lock owner may just decide to stop honoring the lock altogether inside their application

I think this is the clincher. Despite our best efforts at programming fairness here, we end up just adding complexity without preventing this sort of action by the lock owner.

Reply:
I'll go with a default of 20% and allow only lowering for now.  Let me know if you'd rather something else

Reply:
Could we just allow the lock owner to change either up or down?

Reply:
Done.  Default 10% with owner option to change.